egs/wsj/s5/steps/libs/nnet3/xconfig/basic_layers.py

```diff
@@ -0,0 +1,787 @@
+# Copyright 2016    Johns Hopkins University (Dan Povey)
+#           2016    Vijayaditya Peddinti
+# Apache 2.0.
+
+""" This module contains the parent class from which all layers are inherited
+and some basic layer definitions.
+"""
+
+from __future__ import print_function
+import sys
+import libs.nnet3.xconfig.utils as xutils
+from libs.nnet3.xconfig.utils import XconfigParserError as xparser_error
+
+
+class XconfigLayerBase(object):
+    """ A base-class for classes representing layers of xconfig files.
+        This mainly just sets self.layer_type, self.name and self.config/
```

Comment:
I'd remove  "This mainly just sets self.layer_type, self.name and self.config", as the base-class has a bunch of functions... that comment was outdated I think.